pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1652,7 +1652,11 @@ public RealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableConf
 
     // Acquire semaphore to create stream consumers
     try {
-      _partitionGroupConsumerSemaphore.acquire();
+      Instant acquireStart = Instant.now();
+      while (!_partitionGroupConsumerSemaphore.tryAcquire(5, TimeUnit.MINUTES)) {
+        _segmentLogger.warn("Failed to acquire partitionGroupConsumerSemaphore in: {} . Retrying.",
+            Duration.between(acquireStart, Instant.now()));
+      }
```

Comment:
(minor) We usually use `System.currentTimeMillis()` which should have smaller overhead
```suggestion
      long startTimeMs = System.currentTimeMillis();
      while (!_partitionGroupConsumerSemaphore.tryAcquire(5, TimeUnit.MINUTES)) {
        _segmentLogger.warn("Failed to acquire partitionGroupConsumerSemaphore in: {}ms. Retrying.", System.currentTimeMillis() - startTimeMs);
      }
```